### PR TITLE
Bump cel-go to v0.17.7 and introduce set ext library with new options

### PR DESCRIPTION
**Pull request by cici37**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Bump cel-go to v0.17.7 and introduce set ext library with new options.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by TristonianJones — APPROVED** [review]

**Comment by cici37** [comment]
/assign @jpbetz @liggitt 
This PR is to introduce sets extension library into Kubernetes together with the new options cel added for better control changes on versions. Please have a look when you have time :)

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/celcoststability_test.go [hunk 1]
```diff
@@ -241,6 +241,12 @@ func TestCelCostStability(t *testing.T) {
 				"!('x' in self.val1)":                            6,
 				"self.val1 + self.val2 == ['a', 'b', 'c']":       6,
 				"self.val1 + ['c', 'd'] == ['a', 'b', 'c', 'd']": 4,
+				"sets.contains(self.val1, ['a'])":                6,
+				"sets.equivalent(self.val1, ['a', 'b', 'c'])":    21,
+				"sets.intersects(self.val1, ['a'])":              6,
+				"sets.contains(self.val3, [1])":                  1003,
+				"!sets.equivalent(self.val3, [1, 2, 3])":         6004,
+				"sets.intersects(self.val3, [1])":                1003,
```
Comment (jpbetz) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/celcoststability_test.go:249]: Let's have at least 2 tests for each of `contains`, `equivalent` and `intersects`  that test different lengths of inputs (e.g. one that does `[1]` and one that does `[1, 2, 3....]`) and show that the cost changes in the way we expect for each function

Comment (jpbetz) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/celcoststability_test.go:249]: nvm, I missed that the val3 cases use the buildLargeArray input

Comment (cici37) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/celcoststability_test.go:249]: Thanks! I'll resolve this one

**Comment by cici37** [comment]
@liggitt Would you have time for a quick dep approve? Thank you! 🙏

**Comment by cici37** [comment]
Rebased due to conflict.

File: staging/src/k8s.io/apiserver/pkg/cel/environment/base.go [hunk 2]
```diff
@@ -106,6 +108,21 @@ var baseOpts = []VersionedOptions{
 			ext.Strings(ext.StringsVersion(2)),
 		},
 	},
+	// Set library
+	{
+		IntroducedVersion: version.MajorMinor(1, 29),
+		EnvOptions: []cel.EnvOption{
+			ext.Sets(),
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/cel/environment/base.go:115]: Did we not expose any of the sets functions before?

Do we need to update our CEL docs (https://kubernetes.io/docs/reference/using-api/cel/#cel-community-libraries) with the sets library (or would we update that in 1.30 docs)? Is there a tracking issue for updating docs?

**Review by liggitt — COMMENTED** [review]
one question on docs, lgtm otherwise

/approve

File: staging/src/k8s.io/apiserver/pkg/cel/environment/base.go (hunk 2 above)
Comment (cici37) [reply staging/src/k8s.io/apiserver/pkg/cel/environment/base.go:115]: No we didn't expose the sets func before. And YES we need to update our cel doc in 1.30(it is currently in dark mode). I will create an tracking issue for updating docs here: https://github.com/kubernetes/website/issues/43747

**Review by jiahuif — CHANGES_REQUESTED** [review]
/lgtm
Thanks for the cost estimator tests.

**Pull request merged**